server/lorax_server/utils/merges/strategies.py

```diff
@@ -0,0 +1,171 @@
+from abc import ABC
+from collections import defaultdict
+import copy
+from typing import TYPE_CHECKING, Dict, List, Tuple, Type
+
+import torch
+from peft import LoraConfig
+
+from lorax_server.pb.generate_pb2 import (
+    AdapterParameters, 
+    MajoritySignMethod as MajoritySignMethodEnum,
+    MergeStrategy as MergeStrategyEnum,
+)
+from lorax_server.utils.merges.utils import calculate_majority_sign_mask, disjoint_merge, prune
+
+if TYPE_CHECKING:
+    from lorax_server.utils.adapter import ModuleMap
+
+
+def _apply_weights(tensors: List[torch.Tensor], w: torch.Tensor) -> torch.Tensor:
+    t = torch.stack(tensors, dim=0)
+
+    # element-wise weighting of each task tensor
+    # need to unsqueeze weights to match task tensor dimensions
+    # for multiplication to apply element-wise
+    while len(t.shape) > len(w.shape):
+        w = w.unsqueeze(-1)
+    return t * w
+
+
+class MergeStrategy(ABC):
+    def merge(self, task_tensors: List[torch.Tensor], weights: torch.Tensor) -> torch.Tensor:
+        raise NotImplementedError()
+
+
+class LinearMerge(MergeStrategy):
+    def __init__(self, **kwargs):
+        pass
+
+    def merge(self, task_tensors: List[torch.Tensor], weights: torch.Tensor) -> torch.Tensor:
+        weighted_task_tensors = _apply_weights(task_tensors, weights)
+        return weighted_task_tensors.sum(dim=0)
+
+
+class TiesMerge(MergeStrategy):
+    def __init__(self, density: float, majority_sign_method: str = "total", **kwargs):
+        self.density = density
+        self.majority_sign_method = majority_sign_method
+    
+    def merge(self, task_tensors: List[torch.Tensor], weights: torch.Tensor) -> torch.Tensor:
+        # sparsify
+        task_tensors = [prune(tensor, self.density, method="magnitude") for tensor in task_tensors]
+        weighted_task_tensors = _apply_weights(task_tensors, weights)
```

Comment:
Hi @tgaddair, I think this should be fixed similar to what was done in the HF PEFT library. In their initial commit, they had the wrong ordering of the steps. I would suggest to maybe look at the discussion and fixed [here](https://github.com/huggingface/peft/pull/1364/files)

Reply:
Thanks @prateeky2806! Let me put together a quick PR.

Reply:
FYI you would also need to change the order in dare_ties.

Reply:
@prateeky2806 can you take a look at #239 and verify the fix looks correct? Thanks again for pointing this out!

Reply:
Hi @tgaddair, these changes look good, there is just one last thing. It would be ideal to set the default weight for TIES to all ones because that a setting we experimented with in our paper and kind of works well if people do not want to try out different values.

Reply:
I have also left a similar comment in the HF PEFT pull request which I think will be fixed in the coming days.

Reply:
Thanks @tgaddair, this looks correct. I missed this when I was going over the code. 
Everything looks correct to me now.

Reply:
Thanks for verifying @prateeky2806!